src/Core/Language/Pack/Loader/RemoteLanguagePackLoader.php

```diff
@@ -34,35 +36,52 @@
 final class RemoteLanguagePackLoader implements LanguagePackLoaderInterface
 {
     /**
-     * The link from which available languages are retrieved.
+     * The languages repository base path
+     *
+     * @TODO : duplicated from {prestashop_repository}/app/config/config.yml;
+     *         used for initial auto-install (no symfony container available)
      */
-    public const PACK_LINK = 'http://i18n.prestashop.com/translations/%ps_version%/available_languages.json';
+    private const LANG_REPOSITORY_BASE_PATH = 'http://i18n.prestashop-project.org';
 
     /**
      * @var Version
      */
-    private $version;
+    private Version $version;
+
+    /**
+     * @var string represents the language repository base path
+     * @TODO : change type for GuzzleHttp\Client
+     */
+    private string $langRepositoryBasePath;
 
     /**
      * @param Version $version
+     * @param ?string $langRepositoryBasePath should become mandatory in future release
      */
-    public function __construct(Version $version)
+    public function __construct(Version $version, string $langRepositoryBasePath = self::LANG_REPOSITORY_BASE_PATH)
     {
         $this->version = $version;
+        $this->langRepositoryBasePath = $langRepositoryBasePath;
+    }
+
+    /**
+     * @param string $locale if no locale given, return URL for available languages list
+     *
+     * @return string requested URL
+     */
+    public function getLanguagePackUrl(?string $locale = null): string
+    {
+        return $this->langRepositoryBasePath . '/translations/' . $this->version->getSemVersion() . ($locale ? "/{$locale}/{$locale}.zip" : '/available_languages.json');
```

Comment:
You shouldn't use this method for to purposes This one returns both things:
- language list json URL when parameter is empty
- language pack URL when the parameter is provided

Optional parameter that switch the behaviour of a function (this much) are rarely a good idea, it's better to create two distinct methods for those two use cases, and the `$locale` parameter can then be mandatory for `getLanguagePackUrl`

Besides in the list use case the naming `getLanguagePackUrl` has no link with what it actually does

Reply:
```suggestion
    public function getLanguagePackUrl(string $locale): string
    {
        return $this->langRepositoryBasePath . '/translations/' . $this->version->getSemVersion() . sprintf('/%s/%s.zip', $locale}, $locale);
     }

    public function getLanguagePackListUrl(): string
    {
        return $this->langRepositoryBasePath . '/translations/' . $this->version->getSemVersion() . '/available_languages.json');
     }
```

Reply:
Besides the way I used `sprintf` here is not ideal as it could be used to build the whole return not just the last part:

```php
    return sprintf('%s/translations/%s/%s/%s.zip', $this->langRepositoryBasePath, $this->version->getSemVersion(), $locale, $locale);
```